src/main/java/edu/kit/provideq/toolbox/meta/setting/MetaSolverSettingDeserializer.java

```diff
@@ -0,0 +1,56 @@
+package edu.kit.provideq.toolbox.meta.setting;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MetaSolverSettingDeserializer extends JsonDeserializer<MetaSolverSetting> {
+    @Override
+    public MetaSolverSetting deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
+        ObjectCodec codec = jsonParser.getCodec();
+        JsonNode node = codec.readTree(jsonParser);
+
+        // Get type
+        MetaSolverSettingType type = MetaSolverSettingType.valueOf(node.get("type").asText());
+
+        // Create subclass based on the type
+        switch (type) {
+            case CHECKBOX -> {
+                return new Checkbox(
+                        node.get("name").asText(),
+                        node.get("state").asBoolean());
+            }
+            case RANGE -> {
+                return new Range(
+                        node.get("name").asText(),
+                        node.get("min").asDouble(),
+                        node.get("max").asDouble(),
+                        node.get("value").asDouble());
```

Comment:
I don't think the user should be able to overwrite the `min` and `max` values through a request. (Although this might not be the correct place implement a check).

Reply:
Yeah this is just the json deserializer to parse the data from the frontend to the correct format.
The check would need to match the setting defined in the toolbox, ignore anything like a min or max value from the input and just check if the value is within the range.
The proper place to do that would be in the MetaSolver itself.